source/security/key-management.rst

```diff
@@ -39,3 +39,63 @@ Cryptographic keys used for signing |SPN| binaries should be securely managed.
+
+Table below depicts key id’s defined for components and their associated test keys for various key types used for signing components.
+
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| KEY ID                   | RSA2048 KEY                              | RSA3072 KEY                              | Usage                                  |
++=================+========+========================+=================+==========================================+========================================+
+| MASTER_KEY_ID            | MasterTestKey_Priv_RSA2048.pem           | MasterTestKey_Priv_RSA3072.pem           | Signing external key hash store        |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| CFGDATA_KEY_ID           | ConfigTestKey_Priv_RSA2048.pem           | ConfigTestKey_Priv_RSA3072.pem           | Signing CfgData                        |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| FIRMWAREUPDATE_KEY_ID    | FirmwareUpdateTestKey_Priv_RSA2048.pem   | FirmwareUpdateTestKey_Priv_RSA2048.pem   | Signing firmware capsule update        |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| CONTAINER_KEY_ID         | ContainerTestKey_Priv_RSA2048.pem        | ContainerTestKey_Priv_RSA3072.pem        | Signing Container header               |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| CONTAINER_COMP_KEY\_ID   | ContainerCompTestKey_Priv_RSA2048.pem    | ContainerCompTestKey_Priv_RSA3072.pem    | Signing Container component            |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+| OS1_PUBLIC_KEY_ID        | OS1_TestKey_Pub_RSA2048.pem              | OS1_TestKey_Pub_RSA3072.pem              | Public key used to sign Linux OS image |
++--------------------------+------------------------------------------+------------------------------------------+----------------------------------------+
+
+
+One could use either key id or complete path to signing keys while configuring in build scripts.
+
+.. note:: Signing tools support either KEY_ID corresponding to a component or complete path to private key.
+
+Test Keys Generation
+*********************
+
+Keys required for |SPN| can be generated using GenerateKeys.py available
+at BootloaderCorePkg/Tools/. The key generation process could be **one

```

Comment:
the key generation process is a **one time...

Reply:
Done